### PR TITLE
Test#72 new try to add tests

### PR DESCRIPTION
**Pull request by Gymbotan**
Relevant tests from main project were moved here. Some new tests were added.

### CODE REVIEW

File: TuviPgpLibTests/TuviPgpLibTests.csproj [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
```
Comment (ValeraFinebits) [inline TuviPgpLibTests/TuviPgpLibTests.csproj:9]: Please set maximum warnings and analysis level

**Review by ValeraFinebits — APPROVED** [review]

File: TuviPgpLibTests/EccPgpContextTests.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        }
+    }
+
+    internal class EccPgpContextTests
+    {
+        private EccPgpContext InitializeEccPgpContext()
+        {
+            var keyStorage = new MockPgpKeyStorage().Get();
+            var context = new TestEccPgpContext(keyStorage);
+            context.LoadContextAsync().Wait();
```
[the remark this answers is not in the document]
Comment (Gymbotan) [reply TuviPgpLibTests/EccPgpContextTests.cs:40]: It is not a test. It's a private method of context initialization.

File: TuviPgpLibTests/ExternalStorageBasedPgpContextTests.cs [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        }
+    }
+
+    internal class ExternalStorageBasedPgpContextTests
+    {
+        private ExternalStorageBasedPgpContext InitializeTestPgpContext()
+        {
+            var keyStorage = new MockPgpKeyStorage().Get();
+            var context = new TestPgpContext(keyStorage);
+            context.LoadContextAsync().Wait();
```
[the remark this answers is not in the document]
Comment (Gymbotan) [reply TuviPgpLibTests/ExternalStorageBasedPgpContextTests.cs:38]: It is not a test. It's a private method of context initialization.

File: TuviPgpLibTests/TuviPgpContextTests.cs [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+namespace TuviPgpLibTests
+{
+    internal class TuviPgpContextTests
+    {
+        private TuviPgpContext InitializeTuviPgpContext()
+        {
+            var keyStorage = new MockPgpKeyStorage().Get();
+            var context = new TuviPgpContext(keyStorage);
+            context.LoadContextAsync().Wait();
```
[the remark this answers is not in the document]
Comment (Gymbotan) [reply TuviPgpLibTests/TuviPgpContextTests.cs:27]: It is not a test. It's a private method of context initialization.

**Review by anatolse — APPROVED** [review]

File: TuviPgpLibTests/EccPgpContextTests.cs [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            return context;
+        }
+
+        [Test]
+        public void DeterministicEccKeyDerivation()
+        {
+            string ToHex(byte[] data) => string.Concat(data.Select(x => x.ToString("x2")));
+
+            for (int i = 0; i < TestData.EccKeyPairs.Length; i++)
```
[the remark this answers is not in the document]
Comment (Gymbotan) [reply TuviPgpLibTests/EccPgpContextTests.cs:49]: We don't need any parameter here. It's a test to create (derive) keys and check are they equal to determined values. It's work good with for cycle. I can't understand why we should should use exactly TestCaseSource?
Also this type of test allows to add or remove test examples in TestData.

File: TuviPgpLibTests/EccPgpContextTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (al-kau) [reply TuviPgpLibTests/EccPgpContextTests.cs:40]: I mean next:

```
Task <EccPgpContext> InitializeEccPgpContextAsync()
{...}

[Test]
public Task EссEncryptAndDecryptAsync()
{
    using EccPgpContext ctx = await InitializeEccPgpContextAsync();
    ...
}
```

**Review by ValeraFinebits — APPROVED** [review]

**Review by al-kau — APPROVED** [review]

**Pull request merged**